PGM/src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -41,24 +50,95 @@
 import tc.oc.pgm.rotation.RotationState;
 import tc.oc.pgm.teams.TeamFactory;
 
-public class MapCommands {
+public class MapCommands implements Commands {
+    private final UserFinder userFinder;
+    private final SyncExecutor syncExecutor;
+    private final IdentityProvider identityProvider;
+    private final ComponentRenderContext renderer;
+
+    @Inject
+    MapCommands(UserFinder userFinder, IdentityProvider identityProvider, SyncExecutor syncExecutor, ComponentRenderContext renderer) {
+        this.userFinder = userFinder;
+        this.identityProvider = identityProvider;
+        this.syncExecutor = syncExecutor;
+        this.renderer = renderer;
+    }
+
     @Command(
         aliases = {"maplist", "maps", "ml"},
         desc = "Shows the maps that are currently loaded",
-        usage = "[page]",
+        usage = "[-a author] [-g gamemode] [page]",
```

Comment:
I would make this command more "dynamic," so you can combine multiple queries.
`[-a authors] [-g gamemodes] [-p players]`

So the command, `/maps -a=ElectroidFilms -g=ctw,ctf -p=20`, would find a map that I made that is either ctw or ctf and the min/max players fits 20 players.